modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectLayoutResourceImpl.java

```diff
@@ -123,6 +140,31 @@ public ObjectLayout getObjectLayout(Long objectLayoutId) throws Exception {
 			_objectLayoutService.getObjectLayout(objectLayoutId));
 	}
 
+	@Override
+	public ObjectLayout
+			postObjectDefinitionByExternalReferenceCodeObjectDefinitionExternalReferenceCodeObjectLayout(
+				String objectDefinitionExternalReferenceCode,
+				ObjectLayout objectLayout)
+		throws Exception {
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.
+				getObjectDefinitionByExternalReferenceCode(
+					objectDefinitionExternalReferenceCode,
+					contextUser.getCompanyId());
+
+		return _toObjectLayout(
```

Comment:
In the 'getObjectDefinitionByExternalReferenceCodeObjectDefinitionExternalReferenceCodeObjectLayoutsPage' method, you called the method from the resource: `return getObjectDefinitionObjectLayoutsPage(` .
Here, you are calling the Service method. Could you change to call the post method of the Resource?